pkg/server/middleware/introspect.go

```diff
@@ -0,0 +1,122 @@
+package middleware
```

Comment:
I am wondering if it may be confusing to co-locate server code for both binaries. what do you tink about having a duplicate file structure for oidc stuff? or, alliteratively a set of `oidc` packages within existing packages?